esp/smc/SMCLib/TpWrapper.cpp

```diff
@@ -2093,6 +2093,85 @@ extern TPWRAPPER_API void initContainerRoxieTargets(MapStringToMyClass<ISmartSoc
     }
 }
 
+void appendServerAddress(StringBuffer& list, IPropertyTree& env, IPropertyTree& server, const char* farmerPort)
+{
+    //just in case, for backward compatability with old environment.xml files, allow server rather than farmer to specify port
+    const char *port = server.queryProp("@port");
+    if (!port)
+        port = farmerPort;
+    if (port && streq(port, "0")) //0 == roxie listening on queue rather than port
+        return;
+
+    const char *netAddress = server.queryProp("@netAddress");
+    if (!netAddress && server.hasProp("@computer"))
+    {
+        VStringBuffer xpath("Hardware/Computer[@name='%s']/@netAddress", server.queryProp("@computer"));
+        netAddress = env.queryProp(xpath.str());
+    }
+    if (!netAddress || !*netAddress)
+        return;
+    if (list.length())
+        list.append('|');
+    list.append(netAddress).append(':').append(port);
+}
+
+class TlsSmartSocketFactory : public CSmartSocketFactory
+{
+public:
+    TlsSmartSocketFactory(const char *_socklist) : CSmartSocketFactory(_socklist)
+    {
+        tlsService  = true;
```

Comment:
The encapsulation could be better - it is using an implementation detail of the class.  Should this either be an extra parameter to createSmartSocketFactory(), or should it be using createSecureSmartSocketFactory()?

Is there other configuration information (e.g., issuer for mtls) that needs to be provided?